### PR TITLE
[Update](cloud) add inverted index tmp dir support

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

File: be/src/cloud/io/tmp_file_mgr.h [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    TmpFileMgr(const std::vector<TmpFileDirConfig>& configs) {
+        std::for_each(configs.begin(), configs.end(), [this](const TmpFileDirConfig& config) {
+            if (_tmp_file_dirs_size == MAX_TMP_FILE_DIR_SIZE) {
+                LOG(WARNING) << "THE MAX_TMP_FILE_DIR_SIZE is " << MAX_TMP_FILE_DIR_SIZE
+                             << " and ignore the following tmp file dir " << config.path;
+                return;
+            }
+            // FIXME: Error handling
+            auto st = global_local_filesystem()->delete_directory(config.path);
```
Comment (platoneko) [inline be/src/cloud/io/tmp_file_mgr.h:50]: 考虑再搞个 `Status init()` 处理异常？

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 2]
```diff
@@ -209,8 +214,21 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
             return Status::InternalError("init_fulltext_index directory already exists");
         }
 
-        dir = std::unique_ptr<DorisCompoundDirectory>(DorisCompoundDirectoryFactory::getDirectory(
-                _fs, index_path.c_str(), use_compound_file_writer, can_use_ram_dir));
+        if (config::is_cloud_mode()) {
+            _lfs = doris::io::LocalFileSystem::create(
```
Comment (platoneko) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:218]: 没有必要创建 local fs 实例，直接使用 global local fs 传入绝对路径

File: be/src/cloud/config.cpp [hunk 3]
```diff
@@ -43,4 +43,6 @@ DEFINE_mInt32(refresh_s3_info_interval_s, "60");
 DEFINE_mInt32(vacuum_stale_rowsets_interval_s, "300");
 DEFINE_mInt32(schedule_sync_tablets_interval_s, "600");
 
+DEFINE_String(tmp_file_dirs, "${DORIS_HOME}/tmp");
```
Comment (gavinchou) [inline be/src/cloud/config.cpp:46]: Is this format compatible with the original?

**Comment by airborne12** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp [hunk 4]
```diff
@@ -920,7 +920,7 @@ DorisCompoundDirectory* DorisCompoundDirectoryFactory::getDirectory(
     // Write by RAM directory
     // 1. only write separated index files, which is can_use_ram_dir = true.
     // 2. config::inverted_index_ram_dir_enable = true
-    if (config::inverted_index_ram_dir_enable && can_use_ram_dir) {
+    if (can_use_ram_dir) {
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp:923]: why change it?

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

File: be/src/common/config.cpp [hunk 5]
```diff
@@ -908,6 +908,8 @@ DEFINE_Validator(file_cache_type, [](std::string_view config) -> bool {
     return config.empty() || config == "file_block_cache";
 });
 
+DEFINE_String(tmp_file_dir, "tmp");
+
```
Comment (gavinchou) [inline be/src/common/config.cpp:912]: image

it seems not compatible

Comment (airborne12) [reply be/src/common/config.cpp:912]: not using tmp file manager now, just use storage root path tmp directory as temporary writing dir

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**